gips/0080-gip-direct-indexing-payments.md

```diff
@@ -0,0 +1,147 @@
+
+It is important to note that we cannot expect Indexers to take on risk, e.g. by agreeing to a fixed price for a specific subgraph (instead of per unit of work/gas). The cost of indexing each subgraph is unknown until the subgraph has been indexed, so if the price was fixed a rational Indexer would simply stop indexing when the subgraph becomes unprofitable, forcing the payer to start over. Tying payments as much as possible to the verifiable cost of performing the work is the best way to prevent Denial of Service scenarios, just like with gas in Ethereum.
+
+### Terms and Definitions
+
+**Curation**: Curation requires Consumers to stake a certain amount of GRT per subgraph to signal Indexers to provide indexing services.  However, because the strength of the signal depended upon the strength of other signals using GRT, which also had variability, this became a hard mechanism for developers with many subgraphs.
+
+**Indexers:** Indexers are node operators in The Graph Network that stake Graph Tokens (GRT) in order to provide indexing and query processing services. Indexers earn query fees and indexing rewards for their services.
+
+**Indexing Network Payments:**  this represents the payments to Indexers upon verifying successful indexing.  Typically will occur between a Gateway and Indexers selected by their IISA.
+
+**End Consumers:** The core users of data services via Products built onto the Protocol are referred to as Consumers. Specifically, the consumer is the one that receives a benefit (and thus is willing to pay) for data services. In some cases, this may be the developer. In other cases, a consumer may want to leverage the subgraph of another developer. In general, the consumer is the party that interacts with Gateways/Indexers for the indexing services.
+
+**Consumer**: This is referring to one who interacts directly with the Protocol for indexing services.  Often a **Gateway** (see below).
+
+**Gateway:** a permissionless entity which can select an Indexer and pay for Indexing; often associated with a **Studio**
+
+**Studio:** the primary "Product" that End Customers or developers interact with to pay for Indexing and to query Indexed Subgraphs
+
+**Indexing Gas or Unit of Work:** a concept still being defined that will be used to measure the work required to deliver the indexing.  This is likely a combination of compute, storage, and network related “work” needed to be performed by Indexers providing the indexing service.  This could potentially vary by chain.
+
+**Indexing Gas:** this will be the posted “price” by an Indexer in order to perform indexing services based on the unit of work..
+
+**Horizon Escrow:** to trust-minimize the system, as part of the Horizon roll-out, the Gateway will place payments from the Consumer into Horizon Escrow.  The Indexer will be paid from Horizon.  However, there will be a “trusted” version released without Horizon Escrow.
```

Comment:
In the current Horizon code we're calling this "GraphEscrow"

Reply:
also, this is not accurate: "Gateway will place payments from the Consumer into Horizon Escrow" - the Gateway will place its own funds on the Escrow - this may or may not be directly related to payments from end consumers